File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        n = len(nums)
        cnt = [0]* (n+1)
        
        for num in nums:
            if cnt[num] == 0:
                cnt[num] += 1
            else:
                return num
            
        return -1
        

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.java
class Solution {
    public int findDuplicate(int[] nums) {
        int n = nums.length;
        int[] cnt = new int[n+1];
        
        for(int i = 0; i < n; i++){
            if(cnt[nums[i]] == 0) cnt[nums[i]]++;
            else return nums[i];
        }
        return -1;
    }
}